src/main/java/com/uniovi/entities/Incident.java

```diff
@@ -0,0 +1,106 @@
+package com.uniovi.entities;
+import com.uniovi.util.JasyptEncryptor;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+public class Incident {
+	
+	@Id @GeneratedValue
+	private Long id;
+	
+	private String username, password;
+	private String inciName, location;
+	private List<String> tags= new ArrayList<String>();
+	private List<String> moreInfo= new ArrayList<String>();
+
+	public Incident() {}
+	
+	/**
+	 * Minimum initialization of Incident object. None of the parameters can be empty or null
+	 * 
+	 * @param username - of the user that submitted the incident
+	 * @param passw - of the user that submitted the incident
+	 * @param name - of the incident, either descriptive or a code
+	 * @param location - of the incident 
+	 */
+	public Incident(String username, String passw, String name, String location)
+	{
+		if (username=="" || passw=="" ||name=="" || location=="")
+			throw new IllegalArgumentException("Incident fields cannot be empty");
+		
+		this.username = username;
+		this.password = encryptPass(passw);
+		this.inciName = name;
+		this.location = location;
+	}
+	
+	public void addMoreInfo(String property, String value)
+	{
+		moreInfo.add(property +"/"+value);
+	}
+
+	public void setUsername(String username) {
+		this.username = username;
+	}
+
+	public void setPassword(String password) {
+		this.password = encryptPass(password);
```

Comment:
I think we will have a problem here :sweat_smile:. The username and password refer to the Agent submitting the incident, not to the incident. The best solution would be to create again the Agent class and have a OneToMany relationship from agent to incident. Maybe leaving it like this is enough for this project, we will have to see as we continue working on it.
We can leave it like this for now, but we will have to address this problem next week and see what is the best solution.